frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -32,10 +33,57 @@ import { AgentConfig } from '@/types/Agent';
 import { delayInSeconds } from '@/utils/delay';
 import { updateServiceIfNeeded } from '@/utils/service';
 
+/**
+ * Health check hook
+ */
+const useHealthCheck = () => {
+  const { selectedAgentType } = useServices();
+  const { goto: gotoPage } = usePageState();
+
+  return useCallback(
+    async (serviceId?: string) => {
+      if (!serviceId) return true; // TODO: check
+
+      try {
+        const response = await ServicesService.getDeployment(serviceId);
+        const healthcheck = response.healthcheck;
+        if (isEmpty(healthcheck) || !healthcheck.env_var_status?.needs_update) {
+          return true;
+        }
+
+        if (selectedAgentType !== AgentType.Memeooorr) {
```

Comment:
ideally we do it not based on the agent, we specifically agreed to call it `env_var_status.needs_update` to make it agent agnostic.